projects/wave-core-new/src/lib/layers/legend/legend-raster/mapping-raster-legend.component.html

```diff
@@ -1,37 +1,25 @@
 <tbody class="legend">
-    <ng-template [ngIf]="symbology.isUnitUnknown()">
-        <tr>
-            <td>Interpolation</td>
-            <td>{{ interpolationToName(symbology?.unit?.interpolation) }}</td>
-        </tr>
-        <tr>
-            <td>Measurement</td>
-            <td>{{ symbology?.unit?.measurement }}</td>
-        </tr>
-        <tr>
-            <td>Unit</td>
-            <td>{{ symbology?.unit?.unit }}</td>
+    <ng-template [ngIf]="layer.symbology.colorizer.isContinuous()">
+        <tr *ngFor="let breakpoint of layer.symbology.colorizer.getBreakpoints(); let isFirst = first">
+            <td
+                class="gradient"
+                *ngIf="isFirst"
+                [rowSpan]="layer.symbology.colorizer.getBreakpoints().length"
+                [style.background]="layer.symbology.colorizer | waveWappingColorizerToGradient | waveSafeStyle"
+            ></td>
+            <td>{{ breakpoint.value | number: numberPipeParameters }}</td>
+            <td *ngIf="isFirst">{{ (measurement$ | async)?.measurement }}</td>
         </tr>
     </ng-template>
 
-    <ng-template [ngIf]="!symbology.isUnitUnknown()">
-        <tr *ngFor="let breakpoint of symbology.colorizer.breakpoints; let isFirst = first">
-            <ng-template [ngIf]="symbology.isContinuous()">
-                <td
-                    class="gradient"
-                    *ngIf="isFirst"
-                    [rowSpan]="symbology.colorizer.breakpoints.length"
-                    [style.background]="symbology.colorizer | waveWappingColorizerToGradient | waveSafeStyle"
-                ></td>
-                <td>{{ breakpoint.value | number: numberPipeParameters }}</td>
-                <td *ngIf="isFirst">{{ unitToString(symbology?.unit) }}</td>
-            </ng-template>
-            <ng-template [ngIf]="symbology.isDiscrete()">
+    <ng-template [ngIf]="layer.symbology.colorizer.isDiscrete()">
+        <ng-container *ngIf="measurement$ | async | continuousMeasurement as measurement">
```

Comment:
This looks like you should have an observable that has the right data for you instead of using this pipe.

Reply:
it is the right data but not the right type. Maybe it would make more sense to have separate components for continiuous and discrete rasters but for now I think it's ok

Reply:
Something is strange here.
So you have two nested ifs (which could be one).
Then, you check if it is discrete and then cast it to continuous measurement?
I guess you can get rid of the pipe and the double if.